taipy/core/data/in_memory.py

```diff
@@ -98,3 +97,14 @@ def _read(self):
 
     def _write(self, data):
         in_memory_storage[self.id] = data
+
+    def _append(self, data):
+        """Append data to the existing data in the in-memory storage."""
+        if self.id not in in_memory_storage:
+            in_memory_storage[self.id] = []
+
+        if not isinstance(in_memory_storage[self.id], list):
+            in_memory_storage[self.id] = [in_memory_storage[self.id]]
```

Comment:
If `in_memory_storage[self.id]` is not a list, a `NotImplementedError` should be raised.